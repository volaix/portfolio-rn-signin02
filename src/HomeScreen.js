import React from 'react'
import { View, Text, Alert, ImageBackground } from 'react-native'
import { Button, Avatar, FormLabel, FormInput } from 'react-native-elements'
import styled from 'styled-components'
import { OFFWHITE_COLOR, SUBTITLE_COLOR } from './constants';
import LoginModule from './components/LoginModule';

const HeaderText = styled.Text`
  color: ${OFFWHITE_COLOR};
  font-size: 50; 
  font-style: italic;
  font-weight: 200;
  margin-bottom: 30;
`


const imageBackgroundStyle = {
  flex: 1,
  width: '100%',
  height: undefined,
  backgroundColor: 'rgba(0,0,0,0)',
  overflow: 'hidden',
  alignItems: 'center',
  justifyContent: 'center',
}


class HomeScreen extends React.Component {
  onPressGetData = () => {
    console.log('you have press onpressgetdata')
  }

  render() {
    return (
      <ImageBackground
        source={require('./imgs/diego-ph-222506.jpg')}
        style={imageBackgroundStyle}
      >

        <HeaderText>TravelX</HeaderText>

        <LoginModule />

        <Text style={{ marginTop: 20, color: OFFWHITE_COLOR }}>Need Help ?</Text>

      </ImageBackground>
    )
  }
}

export default HomeScreen