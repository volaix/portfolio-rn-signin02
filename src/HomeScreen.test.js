import React from 'react'
import styled from 'styled-components/native'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { shallow, mount, render } from 'enzyme';
import HomeScreen from './HomeScreen';

it('Renders the Screen', () => {
  const tree = renderer.create(
    <HomeScreen />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});