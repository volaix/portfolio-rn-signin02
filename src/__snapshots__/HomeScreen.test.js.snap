// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0)",
      "flex": 1,
      "height": undefined,
      "justifyContent": "center",
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/imgs/diego-ph-222506.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "hsl(000,0%,80%)",
          "fontSize": 50,
          "fontStyle": "italic",
          "fontWeight": "200",
          "marginBottom": 30,
        },
        undefined,
      ]
    }
  >
    TravelX
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "'rgba(0,0,0,0)'",
          "height": undefined,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginBottom": 20,
          "width": "90%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(98,100,124,1)",
            "fontSize": 20,
          }
        }
      >
        Login
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(98,100,124,1)",
            "fontSize": 20,
          }
        }
      >
        Signup
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(34,47,58,1)",
          "borderRadius": 20,
          "height": 230,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 20,
            "marginTop": 30,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "rgba(98,100,124,1)",
              "borderBottomWidth": 1,
              "borderStyle": "solid",
              "flexDirection": "row",
              "height": 40,
              "marginBottom": 10,
              "width": "75%",
            }
          }
        >
          <View
            onPress={undefined}
            style={
              Array [
                false,
                false,
                false,
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            underlayColor="white"
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "hsl(000,0%,70%)",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "marginLeft": 10,
                    },
                  ],
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            placeholder="Username"
            placeholderTextColor="rgba(98,100,124,1)"
            secureTextEntry={undefined}
            style={
              Object {
                "borderBottomColor": "red",
                "color": "hsl(000,0%,70%)",
                "fontSize": 18,
                "marginLeft": 10,
                "width": 180,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "rgba(98,100,124,1)",
              "borderBottomWidth": 1,
              "borderStyle": "solid",
              "flexDirection": "row",
              "height": 40,
              "marginBottom": 10,
              "width": "75%",
            }
          }
        >
          <View
            onPress={undefined}
            style={
              Array [
                false,
                false,
                false,
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
            underlayColor="white"
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "hsl(000,0%,70%)",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "marginLeft": 10,
                    },
                  ],
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            placeholder="Password"
            placeholderTextColor="rgba(98,100,124,1)"
            secureTextEntry={true}
            style={
              Object {
                "borderBottomColor": "red",
                "color": "hsl(000,0%,70%)",
                "fontSize": 18,
                "marginLeft": 10,
                "width": 180,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            Object {
              "borderRadius": 10,
            },
            Object {
              "borderRadius": 10,
            },
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "rgba(147,72,79,1)",
                },
                Object {
                  "borderRadius": 10,
                },
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                Object {
                  "width": 170,
                },
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "hsl(000,0%,80%)",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              LOGIN
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "hsl(000,0%,80%)",
        "marginTop": 20,
      }
    }
  >
    Need Help ?
  </Text>
</View>
`;
